ppmclibs/tinycv_impl.cc

```diff
@@ -228,24 +229,11 @@ std::vector<int> search_TEMPLATE(const Image* scene, const Image* object,
     // Perform the matching. Info about algorithm:
     // http://docs.opencv.org/trunk/doc/tutorials/imgproc/histograms/template_matching/template_matching.html
     // http://docs.opencv.org/modules/imgproc/doc/object_detection.html
-    // Used metric is (sum of) squared differences
     matchTemplate(scene_roi, object_roi, result, CV_TM_SQDIFF);
 
-    // Use error at original location as upper bound
-    Point center = Point(x - scene_x, y - scene_y);
-    double sse = result.at<float>(center);
-    if (sse == 0) {
-        similarity = 1;
-        return { (int)(x), (int)(y) };
-    }
-
-    // We are only interested in points with a similarity > 0
-    //   similarity = .9 + (40 - mse) / 380  ==  0
-    //   <=>  mse = 40 + .9 * 380 = 382
-    sse = std::min(sse, (382.0 * object_roi.total()));
```

Comment:
Now you replace an threshold, which is based on similarity and properly weighted by area by an arbitrary value (INT_MAX).

At least keep the SSE as lower bound, a good match *has* to have an error less than the error at the original location.